packages/common/src/api/user.ts

```diff
@@ -129,8 +136,19 @@ const userApi = createApi({
           encodedDataMessage,
           encodedDataSignature
         })
+        // When fetching withdrawals, get the root account to display
+        // additional transaction context
+        if (
+          type === full.GetUSDCTransactionsTypeEnum.Transfer &&
+          method === full.GetUSDCTransactionCountMethodEnum.Send
+        ) {
+          const rootSolanaAccount = (
+            await getRootSolanaAccount(context.audiusBackend)
+          ).publicKey.toString()
 
-        return data.map(parseTransaction)
+          return data.map(makeParseTransaction(rootSolanaAccount))
```

Comment:
I don't love it, but I think it should probably be done at this layer

Reply:
It's not the worst. Could probably use a little more commentary to understand what's going on. Maybe as a jsdoc on the parsing function?